app/background.py

```diff
@@ -46,22 +46,39 @@ def put(self, data):
 
 
 def background(inputQueue, outputQueue):
+  def redrawProgram(program):
+    """
+    Sends a SIGUSR1 signal to the current program and draws its screen.
+
+    Args:
+      program (CiProgram): an instance of the CiProgram object.
+
+    Returns:
+      None.
+    """
+    pid = os.getpid()
+    signalNumber = signal.SIGUSR1
+    program.render()
+    outputQueue.put(app.render.frame.grabFrame())
+    os.kill(pid, signalNumber)
+
   block = True
-  pid = os.getpid()
-  signalNumber = signal.SIGUSR1
   while True:
     try:
       try:
-        program, message = inputQueue.get(block)
+        program, message, callerSemaphore = inputQueue.get(block)
```

Comment:
Does passing an empty list for message also cause a redraw?

Reply:
Ahh it does. Then I guess we could remove the 'redraw' message.